lib/publiccloud/instance.pm

```diff
@@ -412,16 +414,15 @@ sub wait_for_ssh {
     }    # endif
 
     # check also remote system is up and running:
+    #   SSH host key is not checked and master socket is not used
     my $retry = 0;    # count retries of unexpected sysout
     if ($args{systemup_check} and isok($exit_code)) {
-        # Install server's ssh publicckeys to prevent authentication interactions
-        # or instance address changes during VM reboots.
-        script_run("ssh-keyscan $args{public_ip} | tee -a ~/.ssh/known_hosts /home/$testapi::username/.ssh/known_hosts");
         while (($duration = time() - $start_time) < $args{timeout}) {
             # timeout recalculated removing consumed time until now
             # We don't support password authentication so it would just block the terminal
             $sysout = $self->ssh_script_output(cmd => 'sudo systemctl is-system-running',
-                timeout => $args{timeout} - $duration, proceed_on_failure => 1, username => $args{username});
+                timeout => $args{timeout} - $duration, proceed_on_failure => 1, username => $args{username},
+                ssh_opts => $self->ssh_opts() . ' -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ControlPath=none');
```

Comment:
I could improve visibility and future management if we move this ssh_opts  in a variable outside the loop, i.e. just before it begins:
```suggestion
          ...
          my $opts = $self->ssh_opts() . ' -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ControlPath=none');
          while (...) {
              ... ssh_opts => $opts ...          
          } # end loop
          ...
```

Reply:
Done. Please mark as resolved if you like it.